docs/en/stack/data-frames/dataframe-examples.asciidoc

```diff
@@ -0,0 +1,362 @@
+        "avg" : 2.0
+      },
+      "customer_id" : "10",
+      "user" : "recip",
+      "order_id" : {
+        "value_count" : 59.0
+      },
+      "taxful_total_price" : {
+        "avg" : 66.89790783898304,
+        "sum" : 3946.9765625
+      },
+      "products" : {
+        "product_id" : {
+          "cardinality" : 116.0
+        }
+      }
+    },
+    ...
+  }
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+In the above example we saw how to transform order data into a customer centric 
+index. Doing this makes it easier to answer questions such as:
+
+* Which customers spend the most?
+
+* Which customers spend the most per order?
+
+* Which customers order most often?
+
+* Which customers ordered the least number of different products?
+
+It's possible to answer these questions using aggregations alone, however 
+{dataframes} allow us to persist this data as a customer centric index. This 
+enables us to analyze data at scale and gives more flexibility to explore and 
+navigate data from a customer centric perspective. In some cases, it can even 
+make creating visualizations much simpler.
+
+[float]
+[[example-airline]]
+=== Finding air carriers with the most delays
+
+We use the Flights sample dataset to find out which air carrier delayed the most.
+First, we filter the source data such that excludes all the cancelled flights by 
+using a query filter, then transform the data to contain the distinct number of 
+flights, the sum of delayed minutes, and the sum of the flight minutes by air 
+carrier. Finally, we use a `bucket_script` to determine what percentage of the 
+flight time was actually delay.
+
+[source,js]
+----------------------------------
+POST _data_frame/transforms/_preview
+{
+  "source": {
+    "index": "kibana_sample_data_flights",
+    "query": { <1>
+      "bool": {
+        "filter": [
+          { "term":  { "Cancelled": false } }
+        ]
+      }
+    }
+  },
+  "dest" : {
+    "index" : "sample_flight_delays_by_carrier"
+  },
+  "pivot": {
+    "group_by": { <2>
+      "carrier": { "terms": { "field": "Carrier" }}
+    },
+    "aggregations": {
+      "flights_count": { "value_count": { "field": "FlightNum" }},
+      "delay_mins_total": { "sum": { "field": "FlightDelayMin" }},
+      "flight_mins_total": { "sum": { "field": "FlightTimeMin" }},
+      "delay_time_percentage": { <3>
+        "bucket_script": {
+          "buckets_path": {
+            "delay_time": "delay_mins_total.value",
+            "flight_time": "flight_mins_total.value"
+          },
+          "script": "(params.delay_time / params.flight_time) * 100"
+        }
+      }
+    }
+  }
+}
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+<1> Filter the source data to select only flights that were not cancelled.
+<2> The data is grouped by the `Carrier` field which contains the airline name.
+<3> This `bucket_script` performs calculations on the results that returned by 
```

Comment:
I think since we link to the scripted_metric reference material in the third example, we should link to https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-pipeline-bucket-script-aggregation.html here too.

Reply:
I added the link to the first paragraph of the example as I did the same in the third one.